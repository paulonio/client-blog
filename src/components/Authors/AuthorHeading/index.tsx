import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import Socials from '@/components/Socials';

import { Author } from '@/types/types';

import { sen } from '@/styles/fonts';
import styles from './styled.module.scss';
import { stringToKey } from '@/utils/utils';

interface AuthorsHeadingProps {
  author: Author;
}

const AuthorsHeading: FC<AuthorsHeadingProps> = ({ author }) => {
  const { t } = useTranslation(['authors', 'blog', 'common']);
  const { fullName, urlToAvatar, bio } = author;

  return (
    <section className={styles.wrapper}>
      <div className={styles.heading}>
        <div className={styles.image}>
          <img src={urlToAvatar} alt={fullName} />
        </div>
        <div className={styles.heading__wrapper}>
          <h1 className={`${styles.title} ${sen.className}`}>
            {t('page.greeting', { name: t(stringToKey(fullName), { ns: 'common' }) })}
          </h1>
          <p className="body-s">{t(stringToKey(bio), { ns: 'blog' })}</p>
          <Socials theme="dark" />
        </div>
      </div>
      <div className={styles.patterns__wrapper}>
        <div className={styles.pattern__left} />
        <div className={styles.pattern__right} />
      </div>
    </section>
  );
};

export default AuthorsHeading;
