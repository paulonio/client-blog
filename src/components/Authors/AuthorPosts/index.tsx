import React, { FC } from 'react';
import { useTranslation } from 'next-i18next';

import Posts from '@/components/BlogPosts/Posts';

import { sen } from '@/styles/fonts';
import type { PostType } from '@/types/types';

import styles from './styled.module.scss';

interface AuthorPostsProps {
  posts: PostType[];
}

const AuthorPosts: FC<AuthorPostsProps> = ({ posts }) => {
  const { t } = useTranslation('authors');

  return (
    <section className={styles.wrapper}>
      <h2 className={`${styles.title} ${sen.className}`}>{t('page.title')}</h2>
      <Posts posts={posts} />
    </section>
  );
};

export default AuthorPosts;
