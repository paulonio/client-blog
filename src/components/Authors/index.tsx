import React, { FC } from 'react';
import { sen } from '@/styles/fonts';
import { Author } from '@/types/types';

import Socials from '@/components/Socials';

import styles from './styled.module.scss';

interface AuthorsProps {
  authors: Author[];
}

const Authors: FC<AuthorsProps> = ({ authors }) => {
  return (
    <section className={styles.authors}>
      <h2 className={sen.className}>List of Authors</h2>
      <div className={styles.authors__content}>
        {authors.map(({ fullName, occupation, company, urlToAvatar }) => (
          <div className={styles.author}>
            <div className={styles.author__image}>
              <img src={urlToAvatar} alt={fullName} />
            </div>
            <div className={styles.author__bio}>
              <h3 className={sen.className}>{fullName}</h3>
              <p className="body-l">
                {occupation} @{company}
              </p>
            </div>
            <Socials theme="dark" />
          </div>
        ))}
      </div>
    </section>
  );
};

export default Authors;
