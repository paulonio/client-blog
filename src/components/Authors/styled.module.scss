@import '../../styles/sass/constants';

.authors {
  margin-top: 128px;

  @media screen and (max-width: 768px) {
    margin-top: 32px;
  }

  & h2 {
    text-align: center;
  }

  &__content {
    margin-top: 48px;
    display: flex;
    align-items: stretch;
    flex-wrap: wrap;
    justify-content: space-between;
    gap: 32px;

    @media screen and (max-width: 1200px) {
      justify-content: center;
    }
  }
}

.author {
  width: 23%;
  max-width: 296px;
  padding: 40px 28px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $color-light-l;
  transition: background-color 0.3s;
  &:hover {
    background-color: $color-light-s;
    cursor: pointer;
  }
  

  &__image {
    margin-bottom: 20px;

    & img {
      border-radius: 50%;
    }
  }

  &__bio {
    text-align: center;
  }

  & ul {
    margin-top: 20px;
  }

  @media screen and (max-width: 1200px) {
    width: 33%;
    max-width: 315px;
  }

  @media screen and (max-width: 1040px) {
    width: 50%;
  }

  @media screen and (max-width: 686px) {
    width: 100%;
  }
}