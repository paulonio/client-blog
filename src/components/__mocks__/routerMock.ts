import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

(useRouter as jest.Mock).mockReturnValue({ query: {} });
