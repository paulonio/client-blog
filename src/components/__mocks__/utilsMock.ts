import { isLinkActive, parseString } from '@/utils/utils';

jest.mock('../../utils/utils', () => ({
  isLinkActive: jest.fn(),
  parseString: jest.fn(),
}));

(isLinkActive as jest.Mock).mockReturnValue(true);
(parseString as jest.Mock).mockReturnValue('');
