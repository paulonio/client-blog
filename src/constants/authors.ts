import { Author } from '@/types/types';

export const AUTHORS: Author[] = [
  {
    fullName: 'Floyd Miles',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Floyd.jpg',
  },
  {
    fullName: 'Dianne Russell',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Dianne.jpg',
  },
  {
    fullName: 'Jenny Wilson',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Jenny.jpg',
  },
  {
    fullName: 'Leslie Alexander',
    occupation: 'Content Writer',
    company: 'Company',
    urlToAvatar: '/assets/images/Leslie.jpg',
  },
];
