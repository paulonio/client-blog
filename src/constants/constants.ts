import BusinessIcon from '@/icons/BusinessIcon';
import CyborgIcon from '@/icons/CyborgIcon';
import EconomyIcon from '@/icons/EconomyIcon';
import ShuttleIcon from '@/icons/ShuttleIcon';

import { About, Category } from '@/types/types';

export const ABOUT_US: About = {
  title: 'We are a community of content writers who share their learnings',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
};

export const OUR_MISSION: About = {
  title: 'Creating valuable content for creatives all around the world',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
};

export const CATEGORY_ITEMS: Category[] = [
  {
    title: 'Business',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: BusinessIcon,
  },
  {
    title: 'Startup',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: ShuttleIcon,
  },
  {
    title: 'Economy',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: EconomyIcon,
  },
  {
    title: 'Technology',
    description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit.',
    Icon: CyborgIcon,
  },
];

export const SPECIAL_POST: About = {
  title: 'It started out as a simple idea and evolved into our passion',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
};
