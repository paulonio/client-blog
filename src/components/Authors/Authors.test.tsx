import React from 'react';
import { render } from '@testing-library/react';
import Authors from '.';
import { AUTHORS } from '@/constants/authors';
import '../__mocks__/intersectionObserverMock';

describe('Authors', () => {
  it('should render authors component', () => {
    const { container } = render(<Authors authors={AUTHORS} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with empty array', () => {
    const { container } = render(<Authors authors={[]} />);

    expect(container).toMatchSnapshot();
  });
});
