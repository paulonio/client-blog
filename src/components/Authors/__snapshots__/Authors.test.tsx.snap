// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authors should render authors component 1`] = `
<div>
  <section
    class="authors"
  />
</div>
`;

exports[`Authors should render with empty array 1`] = `
<div>
  <section
    class="authors"
  />
</div>
`;
